### PR TITLE
chore(gitignore): unignore src/bin and remove unimportant files 

### PR DESCRIPTION
**Pull request by YoganshSharma**
# Pull Request

## Description
Allow tracking `src/bin/` while continuing to ignore generated root-level `bin/` artifacts.

`bin` was added to `.gitignore` in #397  to avoid committing downloaded Bitcoin Core binaries. However, the broad `bin` ignore rule also unintentionally ignores files inside `src/bin/`, which contains critical Rust binary entrypoint files.

This PR also removes: `/docs/fidelity_bonds.md` from `.gitignore`
If contributors want to keep local modifications to docs/fidelity_bonds.md untracked, that rule should be added locally in: `git/info/exclude` instead of the shared repository .gitignore.

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code refactor / performance improvement
- [ ] Documentation update only
- [ ] CI / Docker / Build changes
- [x] Other (please describe):
.gitignore cleanup

## Protocol Version(s) Affected
- [ ] Legacy (ECDSA — `messages.rs`, `contract.rs`, `handlers.rs`)
- [ ] Taproot-Musig2 (`messages2.rs`, `contract2.rs`, `handlers2.rs`)
- [ ] Both
- [ x] Neither (infrastructure / tooling only)

> **Note:** When modifying protocol logic, changes must be made to **both** versions unless the
> scope is explicitly version-specific.

## Affected Component(s)
- [ ] **makerd** (background daemon)
- [ ] **taker** (CLI client)
- [ ] **maker-cli** (command-line tool)
- [ ] Core protocol / cryptography / library
- [ ] **watch_tower** (contract monitoring & recovery)
- [ ] Docker / deployment scripts
- [ ] Tests / test framework
- [ ] Documentation (`docs/`)
- [x] Other (please specify):
.gitignore cleanup

### Code Quality
- [ ] I ran `cargo +nightly fmt --all` and committed the result
- [ ] I ran `cargo +stable clippy --all-features --lib --bins --tests -- -D warnings` with zero warnings
- [ ] I ran `cargo +stable clippy --examples -- -D warnings` with zero warnings
- [ ] I ran `RUSTDOCFLAGS="-D warnings" cargo +nightly doc --all-features --document-private-items --no-deps` with zero warnings
- [ ] Pre-commit git hook passes (`ln -s ../../git_hooks/pre-commit .git/hooks/pre-commit` if not already set)

### Testing
- [ ] All unit tests pass (`cargo test`)
- [ ] Integration tests pass (`cargo test --features integration-test`)
- [ ] Changes were manually tested on **regtest**
- [ ] End-to-end maker ↔ taker swap tested (where applicable)
- [ ] Test-only code is gated behind `#[cfg(feature = "integration-test")]`

### Documentation
- [ ] Relevant files in the `docs/` folder were updated
- [ ] Complex logic is commented

### Security & Privacy (Critical)
- [x] This change preserves **trustlessness** and **atomic swap guarantees**
- [x] No regression in **sybil resistance** or **fidelity bond** logic
- [x] Tor anonymity and P2P message flow were reviewed
- [x] No new attack vectors or trust assumptions introduced
- [x] Edge cases and error handling considered
- [x] ZMQ subscription integrity (raw tx/block feed) is unaffected
- [x] `integration-test` feature flag is not reachable in production code paths

## How to Test

```bash
# Standard swap (good baseline)
cargo test --test standard_swap --features integration-test -- --nocapture

# Abort scenarios
cargo test --test abort1 --features integration-test -- --nocapture
cargo test --test taproot_taker_abort1 --features integration-test -- --nocapture

# Malicious behavior & recovery
cargo test --test malice1 --features integration-test -- --nocapture
cargo test --test taproot_timelock_recovery --features integration-test -- --nocapture

# Or run the full suite
cargo test --features integration-test
```

## Summary by CodeRabbit

* **Chores**
  * Updated `.gitignore` configuration to improve build artifact management and exclude additional development environment files and generated artifacts.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,14 +1,14 @@
 /target
 *.swp
-/docs/fidelity_bonds.md
 Cargo.lock
 .vscode
 .idea
 .ra-target
 bin
+!/src/bin
```
Comment (mojoX911) [inline .gitignore]: We can't ignore this. Rest are fine.

**Review by mojoX911 — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (YoganshSharma) [reply .gitignore]: The `bin` in the above line is ignoring src/bin too.
this line unignores that

Comment (mojoX911) [reply .gitignore]: Ah I see. In that case, we should just add the `/bin` pattern. instead of adding a `!` of `/src/bin`.

**Review by mojoX911 — CHANGES_REQUESTED** [review]

**Review by mojoX911 — APPROVED** [review]
Ack
